File: designPattern/factory/javascript/TvFactory.js
import Factory from "./Factory";
import TvProduct from "./TvProduct";

export default class TvFactory extends Factory{

	constructor(){
		super();
		this.tvs = new Array();
	}

	createProduct(compnay, resoluation){
		return new TvProduct(compnay, resoluation);
	}

	registerProduct(product){
		this.tvs.push(product);
	}
}

File: designPattern/factory/python/TvFactory.py
from Factory import Factory
from TvProduct import TvProduct

class TvFactory(Factory):
	def __init__(self):
		self.tvs = []

	def createProduct(self, company, quality):
		return TvProduct(company, quality)

	def registProduct(self, product):
		self.tvs.append(product)